spyder/api/plugins.py

```diff
@@ -268,6 +268,12 @@ class SpyderPluginWidget(PluginWidget):
     # Status: Optional
     shortcut = None
 
+    # Registered file extesions that the plugin can open
+    # inside it.
+    # Example: ['.ipynb'] for spyder-notebook
+    # Status: Optional
+    registered_file_extensions = []
```

Comment:
Let's make it immutable? a tuple perhaps?

Reply:
The only way this could be modified at runtime is if one of the core Spyder plugins changes it. So I don't see the need to make immutable.

Reply:
Good practices :-p